arlo_server/__init__.py

```diff
@@ -538,6 +539,8 @@ def audit_basic_update(election_id):
     db.session.query(Contest).filter_by(election_id=election.id).delete()
 
     for contest in info["contests"]:
+        check_required_fields_for_contest(contest)
```

Comment:
I just noticed that further down in this function, there's a different error reporting scheme for another validation concern. It calls `db.session.rollback()` before reporting the error. Is that generally necessary? Do we need to think about that here?

Reply:
I think we don’t want to leave the database in an inconsistent state, so if we mutate the db before we can do all validation then I think a rollback is appropriate.

Reply:
I like the explicit rollback. I believe it's implied if you don't call `commit`, but yeah let's be explicit.